src/Npgsql/TypeMapping/TypeMapperBase.cs

```diff
@@ -96,6 +97,30 @@ public INpgsqlTypeMapper MapComposite<T>(string? pgName = null, INpgsqlNameTrans
             }.Build());
         }
 
+        public INpgsqlTypeMapper MapComposite(Type compType, string pgName, INpgsqlNameTranslator? nameTranslator = null)
+        {
+            if (string.IsNullOrWhiteSpace(pgName))
+                throw new ArgumentException("pgName can't be empty", nameof(pgName));
+
+            if (nameTranslator == null)
+                nameTranslator = DefaultNameTranslator;
+
+            Type thfType = typeof(CompositeTypeHandlerFactory<>);
+
+            var thf = (NpgsqlTypeHandlerFactory)Activator.CreateInstance(
+                thfType.MakeGenericType(compType),
+                BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic,
+                binder: null,
+                args: new object[] { nameTranslator },
+                culture: null)!;
+
+            return AddMapping(new NpgsqlTypeMappingBuilder {
+                PgTypeName = pgName,
+                ClrTypes = new[] { compType },
+                TypeHandlerFactory = thf,
+            }.Build());
+        }
+
```

Comment:
Let's add non-generic `UnmapComposite` since the API is symmetric.

Reply:
added